### ISSUE TITLE
ACME is broken?

### ISSUE THREAD

**Issue by timeos** [processone/ejabberd#3035]
## Environment

- ejabberd version: 18.09
- Erlang version: Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 10.3.4
- OS: Linux (CentOS Linux release 7.7.1908 (Core))
- Installed from: official rpm

## Bug description

We used ACME for a while and worked fine until the mid of summer (sooner/later, not know exact date) when it stopped working - "authorization" error:

/opt/ejabberd-19.08/bin/ejabberdctl renew_certificates
Error with one or more certificates
  Error for domain: "example.com",  with reason: 'authorization'

 in log I can see this:

```
2019-09-25 08:19:39.540 [error] <0.764.0>@ejabberd_acme:get_config_ca_url:1148 No CA url has been specified in configuration
2019-09-25 08:19:39.560 [info] <0.764.0>@acme_challenge:register_hooks:40 Registering hook for ACME HTTP headers
2019-09-25 08:19:41.454 [error] <0.764.0>@ejabberd_acme_comm:failed_http_request:401 Unexpected status code from <https://acme-v01.api.letsencrypt.org/acme/challenge/418879972/iL_wIQ>: 404, Body: {
  "type": "urn:acme:error:malformed",
  "detail": "No such challenge",
  "status": 404
}
2019-09-25 08:19:41.454 [error] <0.764.0>@ejabberd_acme:create_new_authorization:318 Error: {throw,{error,{unexpected_code,404,"Not Found"}}} getting an authorization for domain: <<"example.com">>
2019-09-25 08:19:41.454 [info] <0.764.0>@acme_challenge:unregister_hooks:44 Unregistering hook for ACME HTTP headers
```

the problem is that the challenge test request never comes from ACME servers at all. 
If I uncomment following line (which was uncommented before of recent upgrade from 18.01):

   ca_url: "https://acme-v01.api.letsencrypt.org"

then I will get: "Error: create_new_account" on command line immediately.

in log:

2019-09-23 12:17:17.650 [error] <0.1030.0>@ejabberd_acme:ensure_account_exists:366 No existing account
2019-09-23 12:20:17.276 [error] <0.1678.0>@ejabberd_acme:create_new_account:287 Error: {error,badarg} creating an account for contact: <<"mailto:example@example.com">>

thanks for helping on this

**Comment by licaon-kter** [comment]
Do try with ejabberd HEAD from git or wait for 19.09 and retest.

**Comment by zinid** [comment]
Duplicated and fixed.

**Comment by timeos** [comment]
this is duplicate to which ID please? 
where it is fixed? in 19.09 ?

**Comment by zinid** [comment]
> where it is fixed? in 19.09 ?

Yes. The implementation was fully rewritten.
